File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    let obj = {};
    
    for (let i = 0; i < nums.length; i++) {
        if (target - nums[i] in obj) {
            return [obj[target-nums[i]], i] // { 7: 1}
        } else obj[nums[i]] = i; //  {2: 0}
    }
    return Object.values(obj);
};

File: 1-two-sum/1-two-sum.go
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
func twoSum(nums []int, target int) []int {

    mapVal := make(map[int]int)
	result := []int{}

    for index, num := range nums {
        if ele, mapping := mapVal[target - num]; mapping {
            result = []int{ele,index}
        }

        mapVal[num] = index
    }
    return result
}